### PR TITLE
Updated HM/expAll/Switch Messages.

### PR DESCRIPTION
**Pull request by Nathan-MV**
- _Updated HM(Cut, Strength, Surf, Rock Smash and Town Map Fly), expAll and switch pkm messages to match BDSP._

### CODE REVIEW

**Comment by Nathan-MV** [comment]
> I've made this a draft because you're still throwing a bunch of nondescript commits at it.

File: Data/Scripts/011_Battle/001_Battle/003_Battle_ExpAndMoveLearning.rb [hunk 1]
```diff
@@ -162,7 +162,7 @@ def pbGainExpOne(idxParty, defeatedBattler, numPartic, expShare, expAll, showMes
     expGained = expFinal - pkmn.exp
     return if expGained <= 0
     # "Exp gained" message
-    if showMessages
+    if showMessages && !expAll
```
Comment (Maruno17) [inline Data/Scripts/011_Battle/001_Battle/003_Battle_ExpAndMoveLearning.rb:165]: I don't think this should be added. It looks like you're relying on the "Your Pokémon gained Exp. Points!" message (line 47) to cut down on the number of messages shown, but that message is only shown if the player has a non-participant in their party, which isn't necessarily the case. If they don't, but the player does have an Exp. All, no messages about Exp. gain will be shown at all, which is obviously not desirable.

**Comment by Maruno17** [comment]
You shouldn't have put two completely unrelated things into the same PR.

I'm very hesitant to add Defog's overworld effect. I think the main reason it is usable in the overworld (and only in Gen 4) is because fog has an in-battle effect, but that effect isn't coded into Essentials. Given that effect was very unpopular, I don't think many people want it either. Since it doesn't affect battle, fog weather is purely cosmetic, which means it probably shouldn't be removable by the player using Defog. I think it's best to let a plugin deal with Defog's overworld effect.

**Comment by Maruno17** [comment]
In SwSh you can't use Defog in the overworld to get rid of fog weather (even though it does affect battles via Misty Terrain). So that's one game where you can't and one game where you can, which is hardly conclusive. However, BDSP shouldn't really count, because it's a remake and thus not evidence of new games using that mechanic.

Given that there are far more core games which have fog but which do _not_ let you remove it with Defog, I'd say Defog's overworld effect should be opt-in at best. And since I'm not willing to add Settings for every single little thing, and since the Defog code is entirely self-contained, I'd say it should be left to a plugin. If future games work the same as BDSP, I'll reconsider it, but for now there isn't enough evidence that it's worth it.

**Pull request merged**